### PR TITLE
feat: mf-4755 set count of pending tasks as browser action badge

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-4755

### CODE REVIEW

File: packages/mask/background/tasks/pending-tasks.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+async function watchTasks() {
+    const { storage } = PersistentStorages.Web3.createSubScope(`${NetworkPluginID.PLUGIN_EVM}_Message`, {
+        messages: {} as Record<string, ReasonableMessage<Request, Response>>,
+    })
+    await storage.messages.initializedPromise
+    checkMessages(Object.values(storage.messages.value))
+    storage.messages.subscription.subscribe(() => {
+        const messages = Object.values(storage.messages.value)
+        checkMessages(messages)
```
Comment (guanbinrui) [inline packages/mask/background/tasks/pending-tasks.ts]: Use TransactionWatcher.

File: packages/mask/background/tasks/setup.ts [hunk 2]
```diff
@@ -3,3 +3,4 @@ import './setup.hmr.js'
 // NotCancellable tasks here
 import './NotCancellable/PrintBuildFlags.js'
 import './NotCancellable/DatabaseCleanup/DropOldDatabase.js'
+import './pending-tasks.js'
```
Comment (Jack-Works) [inline packages/mask/background/tasks/setup.ts]: please make it either NotCancellable (then you should move the file to NotCancellable folder) or Cancellable (you should learn how other cancellable task is written).

File: packages/mask/background/tasks/pending-tasks.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+function checkMessages(messages: Array<ReasonableMessage<Request, Response>>) {
+    const pendingTasks = messages
+        .filter((x) => x.state === MessageStateType.NOT_DEPEND)
+        .sort((a, z) => a.createdAt.getTime() - z.createdAt.getTime())
+    const length = pendingTasks.length
+    browser.browserAction.setBadgeBackgroundColor({
+        color: '#d32f2f',
+    })
+    browser.browserAction.setBadgeText({
```
Comment (Jack-Works) [inline packages/mask/background/tasks/pending-tasks.ts]: please implement this feature via feature detection.

This feature should support both MV3 API `browser.action` and MV2 `browser.browserAction`. You should prefer MV3 API when existing. (Use `if (browser.action)` to detect).

More information:

https://developer.chrome.com/blog/mv3-actions/#manifest-v3-changes

**Review by Jack-Works — CHANGES_REQUESTED** [review]

File: packages/mask/background/tasks/NotCancellable/PendingTasks.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    })
+}
+
+async function watchTasks() {
+    const { storage } = PersistentStorages.Web3.createSubScope(`${NetworkPluginID.PLUGIN_EVM}_Message`, {
+        messages: {} as Record<string, ReasonableMessage<Request, Response>>,
+    })
+    await storage.messages.initializedPromise
+    checkMessages(Object.values(storage.messages.value))
+    storage.messages.subscription.subscribe(() => {
```
Comment (guanbinrui) [inline packages/mask/background/tasks/NotCancellable/PendingTasks.ts:24]: Please subscribe here: `packages/web3-providers/src/Web3/Base/state/Message.ts`.

File: packages/mask/background/tasks/setup.ts [hunk 5]
```diff
@@ -3,3 +3,4 @@ import './setup.hmr.js'
 // NotCancellable tasks here
 import './NotCancellable/PrintBuildFlags.js'
 import './NotCancellable/DatabaseCleanup/DropOldDatabase.js'
+import './NotCancellable/PendingTasks.js'
```
Comment (Jack-Works) [inline packages/mask/background/tasks/setup.ts:6]: I'd suggest to make it a cancellable task, but the current form is also ok.

File: packages/mask/background/tasks/NotCancellable/PendingTasks.ts (hunk 4 above)
Comment (UncleBill) [reply packages/mask/background/tasks/NotCancellable/PendingTasks.ts:24]: I think it's better to not mix them together.

Comment (guanbinrui) [reply packages/mask/background/tasks/NotCancellable/PendingTasks.ts:24]: It's a part of the Message state.

**Comment by guanbinrui** [comment]
Need approval from @Jack-Works.

**Review by guanbinrui — APPROVED** [review]

**Comment by UncleBill** [comment]
@Jack-Works PTAL

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**